src/main/scala/nl.knaw.dans.easy.deposit/package.scala

```diff
@@ -15,34 +15,62 @@
  */
 package nl.knaw.dans.easy
 
-import java.nio.file.{ FileAlreadyExistsException, Paths }
+import java.nio.file.{ Path, Paths }
 import java.util.UUID
 
-import better.files.StringOps
+import better.files.{ File, StringOps }
 import nl.knaw.dans.bag.DansBag
 import nl.knaw.dans.easy.deposit.docs.StateInfo.State.State
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import org.scalatra.servlet.FileItem
 
 import scala.util.{ Failure, Try }
 import scala.xml._
 
-package object deposit {
+package object deposit extends DebugEnhancedLogging {
```

Comment:
you got so many exceptions now, I would almost advise you to move them into their own file (I mean, just one file dedicated to all the errors your can return (call it `errors.scala`, you'll thank me later 😉)).

Also, don't add `DebugEnhancedLogging` to the package object, but rather to the exception classes that require some extra logging in their constructor.